<ui version="4.0" >
 <class>OptimalViewpointInputParametersFormBase</class>
 <widget class="QWidget" name="OptimalViewpointInputParametersFormBase" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>440</width>
    <height>902</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" >
   <property name="margin" >
    <number>0</number>
   </property>
   <property name="spacing" >
    <number>6</number>
   </property>
   <item>
    <widget class="QGroupBox" name="m_groupBoxVisualization" >
     <property name="sizePolicy" >
      <sizepolicy>
       <hsizetype>5</hsizetype>
       <vsizetype>3</vsizetype>
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title" >
      <string>&amp;Visualization parameters</string>
     </property>
     <layout class="QVBoxLayout" >
      <property name="margin" >
       <number>9</number>
      </property>
      <property name="spacing" >
       <number>6</number>
      </property>
      <item>
       <widget class="QPushButton" name="m_advancedControlsPushButton" >
        <property name="text" >
         <string>Advanced controls</string>
        </property>
        <property name="checkable" >
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QFrame" name="m_advancedControlsFrame" >
        <property name="frameShape" >
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow" >
         <enum>QFrame::Raised</enum>
        </property>
        <layout class="QVBoxLayout" >
         <property name="margin" >
          <number>9</number>
         </property>
         <property name="spacing" >
          <number>6</number>
         </property>
         <item>
          <layout class="QHBoxLayout" >
           <property name="margin" >
            <number>0</number>
           </property>
           <property name="spacing" >
            <number>6</number>
           </property>
           <item>
            <widget class="QCheckBox" name="m_checkBoxShade" >
             <property name="statusTip" >
              <string>If checked, shading will be applied when rendering the volume</string>
             </property>
             <property name="text" >
              <string>S&amp;hade</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="m_computeWithOpacityCheckBox" >
             <property name="text" >
              <string>Compute with opacity</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" >
           <property name="margin" >
            <number>0</number>
           </property>
           <property name="spacing" >
            <number>6</number>
           </property>
           <item>
            <widget class="QLabel" name="m_interpolationLabel" >
             <property name="text" >
              <string>Interpolation</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="m_interpolationComboBox" >
             <property name="currentIndex" >
              <number>0</number>
             </property>
             <item>
              <property name="text" >
               <string>Nearest Neighbour</string>
              </property>
             </item>
             <item>
              <property name="text" >
               <string>Linear (Interpolate &amp; Classify)</string>
              </property>
             </item>
             <item>
              <property name="text" >
               <string>Linear (Classify &amp; Interpolate)</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" >
           <property name="margin" >
            <number>0</number>
           </property>
           <property name="spacing" >
            <number>6</number>
           </property>
           <item>
            <widget class="QCheckBox" name="m_specularCheckBox" >
             <property name="text" >
              <string>Specular</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" >
             <property name="margin" >
              <number>0</number>
             </property>
             <property name="spacing" >
              <number>6</number>
             </property>
             <item>
              <widget class="QLabel" name="m_specularPowerLabel" >
               <property name="text" >
                <string>Power</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QDoubleSpinBox" name="m_specularPowerDoubleSpinBox" >
               <property name="decimals" >
                <number>1</number>
               </property>
               <property name="maximum" >
                <double>128.000000000000000</double>
               </property>
               <property name="value" >
                <double>64.000000000000000</double>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" >
           <property name="margin" >
            <number>0</number>
           </property>
           <property name="spacing" >
            <number>6</number>
           </property>
           <item>
            <widget class="QLabel" name="m_updatePlaneLabel" >
             <property name="text" >
              <string>Update plane</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="m_updatePlaneSpinBox" >
             <property name="wrapping" >
              <bool>true</bool>
             </property>
             <property name="specialValueText" >
              <string>All</string>
             </property>
             <property name="maximum" >
              <number>0</number>
             </property>
             <property name="minimum" >
              <number>-1</number>
             </property>
             <property name="value" >
              <number>-1</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="m_updatePlaneRenderPushButton" >
             <property name="text" >
              <string>Render</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" >
           <property name="margin" >
            <number>0</number>
           </property>
           <property name="spacing" >
            <number>6</number>
           </property>
           <item>
            <widget class="QCheckBox" name="m_clusterCheckBox" >
             <property name="text" >
              <string>Cluster</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" >
             <property name="margin" >
              <number>0</number>
             </property>
             <property name="spacing" >
              <number>6</number>
             </property>
             <item>
              <widget class="QLabel" name="m_clusterFirstLabel" >
               <property name="text" >
                <string>First</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="m_clusterFirstSpinBox" />
             </item>
             <item>
              <widget class="QLabel" name="m_clusterLastLabel" >
               <property name="text" >
                <string>Last</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="m_clusterLastSpinBox" />
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QCheckBox" name="m_readExtentFromFileCheckBox" >
           <property name="text" >
            <string>Read extent from file</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" >
        <property name="margin" >
         <number>0</number>
        </property>
        <property name="spacing" >
         <number>6</number>
        </property>
        <item>
         <widget class="QLabel" name="m_labelNumberOfPlanes" >
          <property name="text" >
           <string>Number of &amp;planes</string>
          </property>
          <property name="buddy" >
           <cstring>m_comboNumberOfPlanes</cstring>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="m_comboNumberOfPlanes" >
          <property name="statusTip" >
           <string>Number of planes to use</string>
          </property>
          <item>
           <property name="text" >
            <string>0</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>3</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>4</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>6</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>8</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>12</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>20</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>42</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>162</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" >
        <property name="margin" >
         <number>0</number>
        </property>
        <property name="spacing" >
         <number>6</number>
        </property>
        <item>
         <widget class="QLabel" name="m_labeVisualizationBlockLength" >
          <property name="text" >
           <string>Block &amp;length</string>
          </property>
          <property name="buddy" >
           <cstring>m_spinBoxVisualizationBlockLength</cstring>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QSpinBox" name="m_spinBoxVisualizationBlockLength" >
          <property name="statusTip" >
           <string>Block length (L) to compute excess entropy</string>
          </property>
          <property name="maximum" >
           <number>6</number>
          </property>
          <property name="minimum" >
           <number>2</number>
          </property>
          <property name="value" >
           <number>4</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" >
        <property name="margin" >
         <number>0</number>
        </property>
        <property name="spacing" >
         <number>6</number>
        </property>
        <item>
         <widget class="QLabel" name="m_labelVisualizationImageSampleDistance" >
          <property name="text" >
           <string>Ima&amp;ge sample distance</string>
          </property>
          <property name="buddy" >
           <cstring>m_doubleSpinBoxVisualizationImageSampleDistance</cstring>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="m_doubleSpinBoxVisualizationImageSampleDistance" >
          <property name="statusTip" >
           <string>Horizontal and vertical distance between rays</string>
          </property>
          <property name="decimals" >
           <number>1</number>
          </property>
          <property name="maximum" >
           <double>512.000000000000000</double>
          </property>
          <property name="minimum" >
           <double>0.100000000000000</double>
          </property>
          <property name="singleStep" >
           <double>0.100000000000000</double>
          </property>
          <property name="value" >
           <double>1.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" >
        <property name="margin" >
         <number>0</number>
        </property>
        <property name="spacing" >
         <number>6</number>
        </property>
        <item>
         <widget class="QLabel" name="m_labelVisualizationSampleDistance" >
          <property name="text" >
           <string>Sample dis&amp;tance</string>
          </property>
          <property name="buddy" >
           <cstring>m_doubleSpinBoxVisualizationSampleDistance</cstring>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="m_doubleSpinBoxVisualizationSampleDistance" >
          <property name="statusTip" >
           <string>Distance between samples on one ray</string>
          </property>
          <property name="decimals" >
           <number>1</number>
          </property>
          <property name="maximum" >
           <double>512.000000000000000</double>
          </property>
          <property name="minimum" >
           <double>0.100000000000000</double>
          </property>
          <property name="singleStep" >
           <double>0.100000000000000</double>
          </property>
          <property name="value" >
           <double>1.000000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QCheckBox" name="m_computeCheckBox" >
        <property name="text" >
         <string>Compute entropy</string>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" >
        <property name="margin" >
         <number>0</number>
        </property>
        <property name="spacing" >
         <number>6</number>
        </property>
        <item>
         <widget class="QLabel" name="m_groupingMethodLabel" >
          <property name="text" >
           <string>Groping method</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="m_groupingMethodComboBox" >
          <item>
           <property name="text" >
            <string>None</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>C</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>C (JS)</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>Splitting C</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>Splitting C (JS)</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" >
        <property name="margin" >
         <number>0</number>
        </property>
        <property name="spacing" >
         <number>6</number>
        </property>
        <item>
         <widget class="QLabel" name="m_similarityThresholdLabel" >
          <property name="text" >
           <string>Similarity threshold</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QDoubleSpinBox" name="m_similarityThresholdDoubleSpinBox" >
          <property name="maximum" >
           <double>1.000000000000000</double>
          </property>
          <property name="singleStep" >
           <double>0.010000000000000</double>
          </property>
          <property name="value" >
           <double>0.500000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" >
        <property name="margin" >
         <number>0</number>
        </property>
        <property name="spacing" >
         <number>6</number>
        </property>
        <item>
         <widget class="QLabel" name="m_newMethodLabel" >
          <property name="text" >
           <string>New method</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QComboBox" name="m_newMethodComboBox" >
          <item>
           <property name="text" >
            <string>None</string>
           </property>
          </item>
          <item>
           <property name="text" >
            <string>2</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="m_newMethodOkPushButton" >
          <property name="text" >
           <string>OK</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QStackedWidget" name="stackedWidget" >
        <property name="sizePolicy" >
         <sizepolicy>
          <hsizetype>5</hsizetype>
          <vsizetype>4</vsizetype>
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="currentIndex" >
         <number>0</number>
        </property>
        <widget class="QWidget" name="page0" />
        <widget class="QWidget" name="page1" >
         <layout class="QHBoxLayout" >
          <property name="margin" >
           <number>0</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item>
           <widget class="QLabel" name="m_newMethod2StepLabel" >
            <property name="text" >
             <string>Step</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="m_newMethod2StepSpinBox" >
            <property name="minimum" >
             <number>1</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="m_newMethod2NormalizedCheckBox" >
            <property name="text" >
             <string>Normalized</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
      <item>
       <widget class="Line" name="line" >
        <property name="orientation" >
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" >
        <property name="margin" >
         <number>0</number>
        </property>
        <property name="spacing" >
         <number>6</number>
        </property>
        <item>
         <widget class="QPushButton" name="m_loadTransferFunctionPushButton" >
          <property name="toolTip" >
           <string>Load transfer function</string>
          </property>
          <property name="text" >
           <string>Load...</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="m_saveTransferFunctionPushButton" >
          <property name="toolTip" >
           <string>Save transfer function</string>
          </property>
          <property name="text" >
           <string>Save...</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QComboBox" name="m_selectEditorComboBox" >
        <item>
         <property name="text" >
          <string>Graphic editor</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Editor by values</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QStackedWidget" name="m_editorsStackedWidget" >
        <property name="currentIndex" >
         <number>0</number>
        </property>
        <widget class="QWidget" name="page0" >
         <layout class="QVBoxLayout" >
          <property name="margin" >
           <number>0</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item>
           <widget class="udg::QTransferFunctionEditorByGradient" native="1" name="m_gradientEditor" >
            <property name="statusTip" >
             <string>Define the transfer function for the volume: left click on free space to add a new point; drag a point to move it; right click on a point to delete it</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="page1" >
         <layout class="QVBoxLayout" >
          <property name="margin" >
           <number>0</number>
          </property>
          <property name="spacing" >
           <number>6</number>
          </property>
          <item>
           <widget class="udg::QTransferFunctionEditorByValues" native="1" name="m_editorByValues" />
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QPushButton" name="m_applyPushButton" >
     <property name="text" >
      <string>Apply</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>udg::QTransferFunctionEditorByGradient</class>
   <extends>QWidget</extends>
   <header>qtransferfunctioneditorbygradient.h</header>
  </customwidget>
  <customwidget>
   <class>udg::QTransferFunctionEditorByValues</class>
   <extends>QWidget</extends>
   <header>qtransferfunctioneditorbyvalues.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>m_selectEditorComboBox</sender>
   <signal>activated(int)</signal>
   <receiver>m_editorsStackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>269</x>
     <y>825</y>
    </hint>
    <hint type="destinationlabel" >
     <x>271</x>
     <y>859</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_advancedControlsPushButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>m_advancedControlsFrame</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>227</x>
     <y>307</y>
    </hint>
    <hint type="destinationlabel" >
     <x>203</x>
     <y>387</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>m_newMethodComboBox</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>stackedWidget</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>241</x>
     <y>708</y>
    </hint>
    <hint type="destinationlabel" >
     <x>241</x>
     <y>739</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
